Detectors/ITSMFT/MFT/condition/include/MFTCondition/DCSConfigUtils.h

```diff
@@ -109,171 +112,153 @@ class DCSConfigUtils
     }
   }
 
-  const std::string& getVersion()
+  std::string& getVersion()
   {
     return mVersion;
   }
 
-  auto& getVersionNameLineInCsv() const
+  int getVersionNameLineInCsv()
   {
-    return mVersionNameLine;
+    return mVerNameLine;
   }
-  auto& getRuConfigAddressLineInCsv() const
+  int getRUConfigAddressLineInCsv()
   {
-    return mRuConfigAddressLine;
+    return mRUConfAddLine;
   }
-  auto& getRuConfigValueLineInCsv() const
+  int getRUConfigValueLineInCsv()
   {
-    return mRuConfigValueLine;
+    return mRUConfValLine;
   }
-  auto& getAlpideConfigAddressLineInCsv() const
+  int getALPIDEConfigAddressLineInCsv()
   {
-    return mAlpideConfigAddressLine;
+    return mALPIDEConfAddLine;
   }
-  auto& getAlpideConfigValueLineInCsv() const
+  int getALPIDEConfigValueLineInCsv()
   {
-    return mAlpideConfigValueLine;
+    return mALPIDEConfValLine;
   }
-  auto& getUbbConfigNameLineInCsv() const
+  int getUBBConfigNameLineInCsv()
   {
-    return mUbbNameLine;
+    return mUBBNameLine;
   }
-  auto& getUbbConfigValueLineInCsv() const
+  int getUBBConfigValueLineInCsv()
   {
-    return mUbbValueLine;
+    return mUBBValLine;
   }
-  auto& getDeadMapLineInCsv() const
+  int getDeadMapLineInCsv()
   {
     return mDeadMapLine;
   }
 
  private:
-  const int mVersionNameLine = 0;
-  const int mAlpideConfigAddressLine = 1;
-  const int mAlpideConfigValueLine = 2;
-  const int mRuConfigAddressLine = 3;
-  const int mRuConfigValueLine = 4;
-  const int mUbbNameLine = 5;
-  const int mUbbValueLine = 6;
+  const int mVerNameLine = 0;
+  const int mALPIDEConfAddLine = 1;
+  const int mALPIDEConfValLine = 2;
+  const int mRUConfAddLine = 3;
+  const int mRUConfValLine = 4;
+  const int mUBBNameLine = 5;
+  const int mUBBValLine = 6;
   const int mDeadMapLine = 7;
 
-  const std::string mVersion;
+  std::string mVersion;
 
-  std::map<const std::string, std::map<const std::string, int>> mNameDictType;
-  std::map<const std::string, int> mNameDictRu;
-  std::map<const std::string, int> mNameDictAlpide;
-  std::map<const std::string, int> mNameDictUbb;
-  std::map<const std::string, int> mNameDictDeadMap;
+  std::map<std::string, std::map<std::string, int>> mNameDict;
+  std::map<std::string, int> mNameDictRU;
+  std::map<std::string, int> mNameDictALPIDE;
+  std::map<std::string, int> mNameDictUBB;
```

Comment:
as far as I can see, these maps are used just to fill the content of the `mNameDict`, are they needed as data members rather than being local to `initDictionary()` (or even just a references extracted there from the `mNameDict`).

Reply:
The mNameDicts are used in member functions in this class. So they are necessary to be data members.

Reply:
I was referring on mNameDictRU, mNameDictALPIDE and mNameDictUBB, which are filled and used only in the initialization.